### PR TITLE
Camera: Do not raise exception if VIDIOC_G_PARM is not supported

### PR DESCRIPTION
**Pull request by otyshchenko1**
The "-ENOTTY" error code for VIDIOC_G_PARM ioctl means that
"streaming parameters" is not supported for the underlaying device,
but we can't just raise exception in that case (in comparison with
frameRateSet() which is called from XENCAMERA_OP_FRAME_RATE_SET
and does not worry us too much) as we are called from configToXen(),
so just throw an equivalent warning and set invalid num/denum values.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>

### CODE REVIEW

**Comment by iusyk** [comment]
i am sorry because of my knowledge lack, but what is the difference between Exception and std::exception ?

File: src/Camera.cpp [hunk 1]
```diff
@@ -471,18 +471,32 @@ void Camera::formatEnumerate()
...
+         if (errno == ENOTTY) {
+             /*
+              * This means that "streaming parameters" is not supported
+              * for the underlaying device, but we can't just raise exception
+              * in that case (in comparison with frameRateSet() which is called
+              * from XENCAMERA_OP_FRAME_RATE_SET) as we are called from
+              * configToXen(), just throw an equivalent warning and set
+              * invalid num/denum values.
+              */
+             LOG(mLog, WARNING) << "Failed to call [VIDIOC_G_PARM] for device "
```
Comment (andr2000) [inline src/Camera.cpp]: WARNING and Failed might not match. s/Failed/Warning, VIDIOC_G_PARM is not supported or something

File: src/Camera.cpp [hunk 2]
```diff
@@ -471,18 +471,32 @@ void Camera::formatEnumerate()
...
+                 << mDevPath << " (" << strerror(errno) << ")";
+             frameRate.numerator = 0;
```
Comment (andr2000) [inline src/Camera.cpp:490]: are there any "invalid" values for that?

**Comment by andr2000** [comment]
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`

**Comment by otyshchenko1** [comment]
> i am sorry because of my knowledge lack, but what is the difference between Exception and std::exception ?

"Exception" is a libxenbe exception described here:
https://github.com/xen-troops/libxenbe/blob/master/include/xen/be/Exception.hpp

File: src/Camera.cpp (hunk 2 above)
Comment (otyshchenko1) [reply src/Camera.cpp:490]: I didn't manage to find pre-defined "invalid" values for that

**Pull request merged**